superdesk/core/http/resource_endpoints.py

```diff
@@ -0,0 +1,302 @@
+                )
+            )
+
+    async def process_get_item_request(
+        self,
+        args: ItemRequestViewArgs,
+        params: None,
+        request: HTTPRequest,
+    ) -> HTTPResponse:
+        """Processes a get single item request"""
+
+        service = get_current_async_app().resources.get_resource_service(self.resource_config.name)
+        item = await service.find_by_id_raw(args.item_id)
+        if not item:
+            raise SuperdeskApiError.notFoundError(
+                f"{self.resource_config.name} resource with ID '{args.item_id}' not found"
+            )
+
+        return HTTPResponse(
+            body=item,
+            status_code=200,
+            headers=(),
+        )
+
+    async def process_post_item_request(self, request: HTTPRequest) -> HTTPResponse:
+        """Processes a create item request"""
+
+        service = get_current_async_app().resources.get_resource_service(self.resource_config.name)
+        payload = await request.get_json()
+
+        if payload is None:
+            raise SuperdeskApiError.badRequestError("Empty payload")
+
+        if isinstance(payload, dict):
+            payload = [payload]
+
+        model_instances = []
+        for value in payload:
+            # Validate the provided item,
+            try:
+                value.setdefault("_id", generate_guid(type=GUID_NEWSML))
+                model_instance = self.resource_config.data_class.model_validate(value)
+                model_instances.append(model_instance)
+            except ValidationError as validation_error:
+                return HTTPResponse(convert_pydantic_validation_error_for_response(validation_error), 403, ())
+
+        ids = await service.create(model_instances)
+        return HTTPResponse(ids, 201, ())
+
+    async def process_patch_item_request(
+        self,
+        args: ItemRequestViewArgs,
+        params: None,
+        request: HTTPRequest,
+    ) -> HTTPResponse:
+        """Processes an update item request"""
+
+        service = get_current_async_app().resources.get_resource_service(self.resource_config.name)
+        payload = await request.get_json()
+
+        if payload is None:
+            raise SuperdeskApiError.badRequestError("Empty payload")
+
+        try:
+            await service.update(args.item_id, payload)
+        except ValidationError as validation_error:
+            return HTTPResponse(convert_pydantic_validation_error_for_response(validation_error), 403, ())
+
+        return HTTPResponse({}, 200, ())
+
+    async def process_delete_item_request(
+        self, args: ItemRequestViewArgs, params: None, request: HTTPRequest
+    ) -> HTTPResponse:
+        """Processes a delete item request"""
+
+        service = get_current_async_app().resources.get_resource_service(self.resource_config.name)
+        original = await service.find_by_id(args.item_id)
+
+        if original:
+            await service.delete({"_id": original.id})
```

Comment:
Side affects like that should go to the data layer (service)

Reply:
yep just wondering if we need to do that delete by lookup instead of passing original to the service, or maybe the resource endpoint should only populate that lookup and call the service, it can raise some not found error if needed

Reply:
@petrjasek Makes sense, thanks. I have made this change, and added a `delete_many` function (for cases where we need to delete multiple items)